analysis/tests/combiners_test.py

```diff
@@ -1117,13 +1216,167 @@ def test_create_accumulator(self, probability_to_keep, metric, expected):
              )),
     )
     def test_compute_metrics(self, acc, expected):
-        combiner = combiners.SumAggregateErrorMetricsCombiner([0.5])
+        combiner = combiners.SumAggregateErrorMetricsCombiner(
+            metrics.AggregateMetricType.SUM, [0.5])
         metric = combiner.compute_metrics(acc)
-        self.assertEqual(expected, metric)
+        assert_aggregate_error_metrics_equals(self, expected, metric)
 
         # Test that no type is np.float64
         self.assertTrue(_check_none_are_np_float64(acc))
 
 
+def assert_aggregate_error_metrics_accumulator_equals(
+        test: parameterized.TestCase,
+        expected: combiners.AggregateErrorMetricsAccumulator,
+        actual: combiners.AggregateErrorMetricsAccumulator):
+    test.assertEqual(expected.num_partitions, actual.num_partitions)
```

Comment:
It feels that using reflection would make things more manageble :), basically we need to compare all fields and there are 3 options:
1.  enum attributes
2.  double attributes
3.  list attributes

What do you think about writing a helper function, which take 2 dataclasses of the same type and do comparison? 
For getting fields you can use [dataclasses.fields](https://docs.python.org/3/library/dataclasses.html#dataclasses.fields) or [dataclasses.asdict](https://docs.python.org/3/library/dataclasses.html#dataclasses.fields) or other methods from `dataclasses` module.

This helper function can be useful in other tests as well.